### PR TITLE
Document how we write acceptance tests

### PR DESCRIPTION
**Pull request by phil-davis**
Update and add to the acceptance test documentation.

This PR is aimed at:

- [x] document the standard grammar forms that we are using to write Given When and Then step text
- [ ] give guidelines for how to write the code for new API acceptance test steps
- [ ] give guidelines for how to write the code for new CLI acceptance test steps
- [x] give guidelines on what makes a "good" test scenario
- [ ] how to write scenarios for bugs https://github.com/owncloud/QA/issues/601

from issue #156

### CODE REVIEW

File: modules/developer_manual/pages/core/acceptance-tests.adoc [hunk 1]
```diff
@@ -3,192 +3,113 @@
...
-
 [[features]]
 === `features/`
 
+This directory contains sub-directories for each of the test suites.
+
+=== `features/suiteName`
 
-This directory stores http://docs.behat.org/en/v2.5/guides/1.gherkin.html[Behat's feature files]. 
+This directory stores http://http://behat.org/en/latest/guides.html[Behat's feature files] for the test suite.
```
[the remark this answers is not in the document]
Comment (phil-davis) [reply modules/developer_manual/pages/core/acceptance-tests.adoc]: cut-paste gone crazy! Fixed.

File: modules/developer_manual/pages/core/acceptance-tests.adoc [hunk 2]
```diff
@@ -300,5 +221,306 @@ If you want to have encryption enabled add `OC_TEST_ENCRYPTION_ENABLED=1`, as in
...
+ * @param string $url
+ */
+public function exampleFunction($method, $url) {
+----
+
+== References
+
+For more information on Behat, and how to write acceptance tests using it, see http://behat.org/en/latest/guides.html[the Behat documentation].
+
+For background information on BDD, see https://dannorth.net/whats-in-a-story/[Dan North resources].
```
Comment (mmattel) [inline modules/developer_manual/pages/core/acceptance-tests.adoc]: Mind to write down what BDD is, else this is just an acronym hard to understand:
Behaviour-driven development (BDD)

File: modules/developer_manual/pages/core/acceptance-tests.adoc [hunk 3]
```diff
@@ -3,193 +3,114 @@
 [[the-test-directory-structure]]
 == The Test Directory Structure
 
-This is the structure of acceptance directory inside 
+This is the structure of the acceptance directory inside
 link:https://github.com/owncloud/core[the core repository's] `tests` directory:
```
Comment (mmattel) [inline modules/developer_manual/pages/core/acceptance-tests.adoc]: link: in front of http is not necessary.
http is its own prefix (link: is only needed in special cases)

File: modules/developer_manual/pages/core/acceptance-tests.adoc (hunk 2 above)
Comment (phil-davis) [reply modules/developer_manual/pages/core/acceptance-tests.adoc]: Done

File: modules/developer_manual/pages/core/acceptance-tests.adoc (hunk 3 above)
Comment (phil-davis) [reply modules/developer_manual/pages/core/acceptance-tests.adoc]: That was existing crud, I fixed it.

**Comment by phil-davis** [comment]
This was work-in-progress. Anyway, what I had so far was a good start! I will pull master and start a new branch and PR for the next parts.

**Comment by mmattel** [comment]
👍

**Pull request merged**